File: 자료구조/C++/백준-스택-10799.cpp
# include <iostream>
# include <stack>

using namespace std;

int main()
{
	ios::sync_with_stdio(false);
	cin.tie(NULL);

	string input;
	cin >> input;

	int result = 0;
	stack<char> str;


	for (int i = 0; i < input.length(); i++)
	{
		if (input[i] == '(')
		{
			str.push(input[i]);
		}
		else if (input[i] == ')' && input[i - 1] == '(')
		{
			str.pop();
			result += str.size();
		}
		else {
			result++;
			str.pop();
		}
	}

	cout << result << endl;
	return 0;
}

File: 자료구조/Python/백준-스택-10799.py
def main():
    str = input()
    stack = []
    total = 0
    for i in range(len(str)):
        if str[i] == '(':
            stack.append(str[i])
        elif str[i] == ')' and str[i-1] == '(':
            stack.pop()
            total += len(stack)
        else:
            total += 1
            stack.pop()
    print(total)

if __name__ == "__main__":
    main()